src/LoginWithAmazon.js

```diff
@@ -1,28 +1,35 @@
 import React from 'react';
 
-// Options variable to request for implicit grant. Logic for assigning 'deviceSerialNumber' needs to be revisited.
+// Options variable to request for implicit grant.
+// TODO: Logic for assigning 'deviceSerialNumber' needs to be revisited.
 const options = {
     scope : ['alexa:all', 'profile'],
     scope_data : {"alexa:all":{"productID":"Silent_Alexa","productInstanceAttributes":{"deviceSerialNumber":"12345"}}}
 };
 export default class LoginWithAmazon extends React.Component {
   render() {
     return (
-        <button type='button' onClick={() => this.handleLogin()}><img alt="demoImage" src="https://images-na.ssl-images-amazon.com/images/G/01/lwa/btnLWA_gold_156x32.png"
+      <button type='button' onClick={() => this.handleLogin()}>
+        <img alt="demoImage" src="https://images-na.ssl-images-amazon.com/images/G/01/lwa/btnLWA_gold_156x32.png"
           width="156" height="32" />
-        </button>
+      </button>
     )
   }
 
   handleLogin() {
+    // This returns an AuthorizeRequest object. After the request is complete, the object will
+    // contain properties detailing the response
     window.amazon.Login.authorize(options, response => this.handleResponse(response));
   }
 
   handleResponse(response) {
-    if(response.error) {
-          console.log("Encountered an error on login: " + response.error)
+    if(typeof response === 'undefined') {
+      console.log("Response is undefined");
+    }
+    else if(response.error) {
+      console.log("Encountered an error on login: " + response.error);
```

Comment:
Does this actually print the response object deeply? I think we need to use some util.

Reply:
I remember writing log lines like this.             

console.log('Failed to fetch new messages for the user: ' + util.inspect(error, { showHidden: true, depth: null }));

See if that applies here.

Reply:
Updated console logs to print the error deeply.

Reply:
Updated the console log to print the response deeply in case of error.